cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/proto/RangeFilter.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.report.proto;
+
+import co.cask.cdap.report.util.ReportField;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a filter that checks whether a given value of a field is within the allowed range.
+ *
+ * @param <T> type of the values
+ */
+public class RangeFilter<T extends Comparable<T>> extends Filter<T> {
+  private final Range<T> range;
+
+  public RangeFilter(String fieldName, Range<T> range) {
+    super(fieldName);
+    this.range = range;
+  }
+
+  /**
+   * @return the allowed range of values of this field
+   */
+  public Range getRange() {
+    return range;
+  }
+
+  @Override
+  public boolean apply(T value) {
+    return (range.getMin() == null || range.getMin().compareTo(value) <= 0)
+      && (range.getMax() == null || range.getMax().compareTo(value) > 0);
+  }
+
+  @Override
+  public List<String> getErrors() {
+    ArrayList<String> errors = new ArrayList<>();
+    errors.addAll(super.getErrors());
```

Comment:
This is not validating fields. It's validating whether the field name in the filter is valid. There can be such case where the user wants to use certain fields to perform filtering, but don't want to include these fields in the final report.

Reply:
rather you can get all unique fields across fields, sort, filter and validate their field names and skip the field name check here, to avoid duplicate checks and error messages.